training/distributed_training/pytorch/model_parallel_v2/llama_v2/smp-train-llama-fsdp-tp-fp8.ipynb

```diff
@@ -0,0 +1,1027 @@
+    "    data_channels = {\"train\": train, \"test\": train}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set Hyperparameters, Metric Definitions, and MPI Options"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### FP8 Datatype\n",
+    "\n",
+    "FP8 (an 8-bit floating point precision) is a datatype suitable for mixed precision training, a natural progression for accelerating deep learning training beyond the 16-bit formats such as FP16 and BF16. NVIDIA H100 GPUs came out supporting mixed precision training with FP8 data types. With P5 instances equipped with H100 GPUs, you can accelerate distributed training with FP8 mixed precision with virtually no accuracy impact.\n",
+    "\n",
+    "For more information, see [FP8 training on P5](https://docs.aws.amazon.com/sagemaker/latest/dg/model-parallel-core-features-v2-fp8-training-on-p5.html).\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# @title Hyperparameters.\n",
+    "\n",
+    "fp8 = 1  # Enable FP8 mixed precision. 1=True, 0=False.\n",
+    "tensor_parallel_degree = 2  # An integer in [1, world_size]\n",
+    "hybrid_shard_degree = (\n",
+    "    4  # # An integer in [0, world_size // tensor_parallel_degree] and its default value is 0.\n",
+    ")\n",
+    "offload_activations = True  # Enables SM activation offloading implementation.\n",
+    "activation_loading_horizon = (\n",
+    "    2  # Activation loading horizon, a positive integer and its default value is 2.\n",
+    ")\n",
+    "save_steps = 10  # Save step interval.\n",
+    "max_steps = 15  # Maximum training steps.\n",
+    "\n",
+    "hyperparameters = {\n",
+    "    \"activation_checkpointing\": 1,\n",
+    "    \"auto_wrap_policy\": \"transformer_auto_wrap_policy\",\n",
+    "    \"backward_fetch_policy\": \"backward_pre\",\n",
+    "    \"beta1\": 0.9,\n",
+    "    \"beta2\": 0.95,\n",
+    "    \"bf16\": 1,\n",
+    "    \"checkpoint_dir\": \"/opt/ml/checkpoints\",\n",
+    "    \"checkpoint_freq\": save_steps,\n",
+    "    \"clean_cache\": 0,\n",
+    "    \"delayed_param\": 1,\n",
+    "    \"enable_memory_profiling\": 0,\n",
+    "    \"epochs\": 100,\n",
+    "    \"fast_validation\": 0,\n",
+    "    \"forward_prefetch\": 1,\n",
+    "    \"fp8\": fp8,\n",
+    "    \"limit_all_gathers\": 1,\n",
+    "    \"logging_freq\": 1,\n",
+    "    \"lr\": 0.0001,\n",
+    "    \"lr_decay_iters\": 47683,\n",
+    "    \"lr_decay_style\": \"cosine\",\n",
+    "    \"max_steps\": max_steps,\n",
+    "    \"min_lr\": 1e-05,\n",
+    "    \"model_type\": model_type,\n",
+    "    \"num_kept_checkpoints\": 2,\n",
+    "    \"plateau\": 0.0,\n",
+    "    \"seed\": 12345,\n",
+    "    \"sharding_strategy\": \"hybrid_shard\",\n",
+    "    \"train_batch_size\": 32,\n",
+    "    \"use_smp_flash_attn\": 1,\n",
+    "    \"val_batch_size\": 32,\n",
```

Comment:
use_smp_implementation=1 ?

Reply:
Oh, thanks for the catch! Looks like it's missing from all notebooks, will update.